signer/cosigner_nonce_cache.go

```diff
@@ -36,6 +34,56 @@ type CosignerNonceCache struct {
 	cache NonceCache
 
 	pruner NonceCachePruner
+
+	movingAverage *movingAverage
+
+	empty chan struct{}
+}
+
+type movingAverageItem struct {
+	timeSinceLastReconcile time.Duration
+	noncesPerMinute        float64
+}
+
+type movingAverage struct {
+	items  []movingAverageItem
+	period time.Duration
+}
+
+func newMovingAverage(period time.Duration) *movingAverage {
+	return &movingAverage{period: period}
+}
+
+func (m *movingAverage) add(
+	timeSinceLastReconcile time.Duration,
+	noncesPerMinute float64,
+) {
+	duration := timeSinceLastReconcile
+	keep := len(m.items) - 1
+	for i, e := range m.items {
+		duration += e.timeSinceLastReconcile
+		if duration >= m.period {
+			keep = i
+			break
+		}
+	}
+	m.items = append(
+		[]movingAverageItem{{timeSinceLastReconcile: timeSinceLastReconcile, noncesPerMinute: noncesPerMinute}},
+		m.items[:keep+1]...,
+	)
+}
+
+func (m *movingAverage) average() float64 {
+	weightedSum := float64(0)
+	duration := float64(0)
+
+	for _, e := range m.items {
+		d := float64(e.timeSinceLastReconcile)
```

Comment:
Won't this result in returning nonces/nanosecond? Maybe this: `d := e.timeSinceLastReconcile.Minutes()`?

Reply:
Actually, the scaling doesn't matter since it cancels itself out.